x-pack/auditbeat/module/system/system_name_agent.go

```diff
@@ -0,0 +1,12 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build agentbeat
+
+package system
+
+const (
+	// ModuleName is the name for this module.
+	ModuleName = "audit/system"
```

Comment:
This breaks the `audit/system` input without another change.

In auditbeat modules are the input names. In the auditbeat spec file we have the same collision, so call this `audit/system` as you do here. However, because prior to this change the actual name of the module was `system` when we received the configuration from agent auditbeat stripped off the `audit` prefix.

https://github.com/elastic/beats/blob/4f46fcb888b7f12706d005b7e1d2d1dafcbe6fcd/x-pack/auditbeat/cmd/root.go#L31-L46

I think you have to stop doing this for the system module (or add the audit prefix to every auditbeat module). The other one in the auditbeat spec file is `audit/file_integrity` https://github.com/elastic/elastic-agent/blob/67ba0aa1869206af14c68a6ecd0a666c1ba3fc57/specs/auditbeat.spec.yml#L40

Reply:
Okay this will need to be conditional then between auditbeat and agentbeat. I will look at doing that, thanks for pointing it out.

 There was a lot of unique stuff per beat, that I learned doing this PR.

Reply:
That `auditbeatCfg` only ever gets invoked when auditbeat is managed by agent.

When agent sends an auditbeat unit configuration, all the modules have an `audit/` prefix, which this code then strips out because auditbeat doesn't expect it.

In the system case we now include the `audit/` prefix, so we should stop stripping it out. It would actually be simpler if we kept the `audit/` prefix on all the modules when running under agentbeat, then we could remove the `strings.Split(rawIn.Type, "/")[1] ` in the code above and it should work.

I don't think you changed anything for Metricbeat but it is the same except that Metricbeat modules have a `/metrics` suffix to strip off instead of an `audit/` prefix. Auditbeat is based on Metricbeat where a module is the equivalent concept to an input in Filebeat.